statistics/handle/storage/read.go

```diff
@@ -127,6 +145,25 @@ func FMSketchFromStorage(sctx sessionctx.Context, tblID int64, isIndex, histID i
 	return statistics.DecodeFMSketch(rows[0].GetBytes(0))
 }
 
+// CheckSkipPartition checks if we can skip loading the partition.
+func CheckSkipPartition(sctx sessionctx.Context, tblID int64, isIndex int, histsID int64) error {
+	rows, _, err := util.ExecRows(sctx, "select distinct_count from mysql.stats_histograms where table_id =%?", tblID)
+	if err != nil {
+		return err
+	}
+	if len(rows) == 0 {
+		return types.ErrPartitionStatsMissing
+	}
+	rows, _, err = util.ExecRows(sctx, "select distinct_count from mysql.stats_histograms where table_id = %? and is_index = %? and hist_id = %?", tblID, isIndex, histsID)
+	if err != nil {
+		return err
+	}
```

Comment:
We can read the `is_index` and `hist_id` in the first SQL, and the filtering?

Reply:
It seems that we will check it for every column and index. It might be a waste when the table is wide.

Reply:
Great, I split this into two parts.